app/scripts/modules/core/src/pipeline/executions/execution/Execution.tsx

```diff
@@ -200,6 +206,38 @@ export class Execution extends React.Component<IExecutionProps, IExecutionState>
     });
   }
 
+  private getDisplayableParameters = memoize((execution: IExecution) => {
```

Comment:
I think `getDisplayableParameters` and `getPinnedParameters` could be simplified.  

It might be worth trying a different approach like (pseudocode):

```
  render() {
    const strategyExclusions = ['parentPipelineId', 'strategy', 'parentStageId', 'deploymentDetails', 'cloudProvider'];
    const paramConfigsByName = keyBy(pipelineConfig.parameterConfig, cfg => cfg.name);  // from lodash

    const isParamDisplayable = (paramName: string) =>
      execution.isStrategy ? !strategyExclusions.includes(name) : true;
    const isParamPinned = (paramName: string) =>
      paramConfigsByName[paramName] && paramConfigsByName[paramName].pinned === true;

    const paramNames = Object.keys(trigger && trigger.parameters || {}).sort();
    const displayableParamNames = allParameters.filter(isParamDisplayable);
    const [pinnedParamNames, unpinnedParamNames] = partition(displayableParamNames, isParamPinned); // from lodash
  }
```

Reply:
shortest I could get after formatting was 
```js
  private getDisplayableParameters = memoizeOne(
    (
      execution: IExecution,
      pipelineConfig: IPipeline,
    ): { displayableParameters: IDisplayableParameter[]; pinnedDisplayableParameters: IDisplayableParameter[] } => {
      // these are internal parameters that are not useful to end users
      const strategyExclusions = [
        'parentPipelineId',
        'strategy',
        'parentStageId',
        'deploymentDetails',
        'cloudProvider',
      ];

      const isParamDisplayable = (paramKey: string) =>
        execution.isStrategy ? !strategyExclusions.includes(paramKey) : true;

      const displayableParameters: IDisplayableParameter[] = Object.keys(
        (execution.trigger && execution.trigger.parameters) || {},
      )
        .filter(isParamDisplayable)
        .sort()
        .map((paramKey: string) => {
          return { key: paramKey, value: JSON.stringify(execution.trigger.parameters[paramKey]) };
        });

      let pinnedDisplayableParameters: IDisplayableParameter[] = [];

      if (pipelineConfig) {
        const paramConfigIndexByName = keyBy(pipelineConfig.parameterConfig, 'name');
        const isParamPinned = (param: IDisplayableParameter): boolean =>
          pipelineConfig.pinAllParameters ||
          (paramConfigIndexByName[param.key] && paramConfigIndexByName[param.key].pinned); // an older execution's parameter might be missing from a newer pipelineConfig.parameterConfig

        pinnedDisplayableParameters = displayableParameters.filter(isParamPinned);
      }

      return { displayableParameters, pinnedDisplayableParameters };
    },
  );
```

which.... it's kind of longer lol, but at least it's just 1 call now.